templates/mon_mysql.sql.j2

```diff
@@ -6,7 +6,7 @@ SET foreign_key_checks = 0;
 CREATE TABLE `alarm` (
   `id` varchar(36) COLLATE utf8mb4_unicode_ci NOT NULL,
   `alarm_definition_id` varchar(36) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
-  `state` enum('UNDETERMINED','OK','ALARM') COLLATE utf8mb4_unicode_ci NOT NULL,
```

Comment:
@rbrndt, We've made some drafts, some investigations and basically it is achievable but effort we would have to put in to make it work is tremendously greater. Not only migration would be harder, more vulnerable but there would be changes in at least monsaca-api and monasca-thresh as there are repositories (DAOs) working with plain MySQL (notice change in columns from varchars/enums to integer). Not the mention that the same thing could happen for other projects using the same database.

@tkuhlman, Do you think that transition to ref tables is something monasca would want to have instead of varchar based columns and triggers to force constraints ?